bin/genesis

```diff
@@ -1291,6 +1291,14 @@ build_bosh_init_manifest() {
 ####################################################
 # multi-call handlers
 
+sha1sum() {
+  if sha1sum 2>/dev/null; then
+    sha1sum "$@"
+  else
+    shasum "$@"
+  fi
+}
+
```

Comment:
This works great if sha1sum isn't installed; the shell errors because it can't exec the command.

This does not work if sha1sum is installed.

Instead of "succeeding", `sha1sum` blocks forever waiting for input from stdin.  A better way to accomplish this (given that this is bash) is to use the `command` builtin.  Specifically, `command -v sha1sum` works like `which sha1sum` except that (A) it doesn't consider alias functions in the calling environment, and (B) it doesn't result in a fork.  If the binary is not in `$PATH`, `command -v ...` returns no output, but if it *is*, it returns the full path to the binary.  You can check the output with `-n` or `-z` and dispatch accordingly.